android-pdf-viewer/src/main/java/com/infomaniak/lib/pdfview/DragPinchManager.java

```diff
@@ -299,9 +314,38 @@ public boolean onTouch(View v, MotionEvent event) {
                 onScrollEnd(event);
             }
         }
+
+        if (hasTouchPriority) {
+            retVal = handleTouchPriority(event, v);
+        }
+
         return retVal;
     }
 
+    private ViewParent getViewToDisableTouch(View startingView) {
+        ViewParent parentView = startingView.getParent();
+        while (!(((View) parentView) instanceof RecyclerView)) {
+            parentView = parentView.getParent();
+        }

```

Comment:
I need to add a security here to avoid an infinite loop. For example: if we reach the root of the view and `getParent()` return `null`, we can break the loop.